helix-view/src/clipboard.rs

```diff
@@ -77,7 +77,11 @@ pub fn get_clipboard_provider() -> Box<dyn ClipboardProvider> {
             copy => "tmux", "load-buffer", "-";
         }
     } else {
-        Box::new(provider::NopProvider)
+        #[cfg(target_os = "windows")]
+        return Box::new(provider::WindowsProvider);
```

Comment:
Need the `return`s here because it gives you a warning otherwise.

Reply:
I think can just remove the return or you could use `if cfg!(target_os = "windows") { ... }`.

Reply:
`if cfg!(target_os = "windows") { ... }` sounds good

Reply:
https://doc.rust-lang.org/rust-by-example/attribute/cfg.html
Unfortunately, `cfg!()` would still compile the code and perform the check at runtime, which means it wouldn't compile for other platforms because `clipboard-win` depends upon `winapi`. This was what I originally did before I realized it didn't work, though the commits are now squashed.

Reply:
Also, the `#[cfg]` macro also seems to want the following line to end in a semi-colon, so I have to make it an explicit `return`.